Library/Homebrew/cmd/readall.rb

```diff
@@ -4,13 +4,17 @@
 # or to determine if any current formulae have Ruby issues
 
 require "formula"
+require "open3"
 require "tap"
 require "thread"
 
 module Homebrew
   def readall
     if ARGV.delete("--syntax")
       ruby_files = Queue.new
+      if (HOMEBREW_LIBRARY/"brew.rb").exist?
```

Comment:
When running the test suite, where `HOMEBREW_LIBRARY` is altered to point somewhere else. But that's addressed by @xu-cheng's suggestion in a more elegant way.